src/main/java/edu/kimjones/advancedjava/stock/StockQuoteApplication.java

```diff
@@ -76,38 +80,62 @@ private void doMain(String args[]) throws IOException {
+            System.out.printf("%nThe latest price of stock %s is %.2f %n", symbol, latestQuote.getStockPrice());
 
-        /**
-         * get a list of price quotes for each day from calFromDate to calUntilDate
-         */
-        List<StockQuote> dailyQuoteList = stockService.getStockQuote(symbol, calFromDate, calUntilDate);
+            System.out.printf("%n***** %n%n");
 
-        for (StockQuote temp : dailyQuoteList) {
-            System.out.printf("The price of stock %s on date %tD is %.2f %n", temp.getStockSymbol(), temp.getDateRecorded(), temp.getStockPrice());
-        }
+            /**
+             * get the a price quote for a stock on a particular date
+             */
 
-        System.out.printf("%n***** %n%n");
+            StockQuote quoteOnDate = stockService.getStockQuote(symbol, fromDate);
+
+            System.out.printf("%nThe price of stock %s on date %tD is %.2f %n", symbol, quoteOnDate.getDateRecorded(), quoteOnDate.getStockPrice());
+
+            System.out.printf("%n***** %n%n");
+
+            /**
+             * get a list of price quotes for each day from calFromDate to calUntilDate
+             */
 
-        StockService.StockQuoteInterval interval = StockService.StockQuoteInterval.EVERY_SIX_HOURS;
-        /**
-         * get a list of price quotes for each day from calFromDate to calUntilDate on interval
-         */
-        List<StockQuote> hourlyQuoteList = stockService.getStockQuote(symbol, calFromDate, calUntilDate, interval);
+            List<StockQuote> dailyQuoteList = stockService.getStockQuoteList(symbol, calFromDate, calUntilDate);
+
+            System.out.printf("%n");
+
+            for (StockQuote temp : dailyQuoteList) {
+                System.out.printf("The price of stock %s on date %tD is %.2f %n", temp.getStockSymbol(), temp.getDateRecorded(), temp.getStockPrice());
+            }
+
+            System.out.printf("%n***** %n%n");
+
+            /**
+             * get a list of price quotes for each day from calFromDate to calUntilDate on interval
+             */
+
+            StockService.StockQuoteInterval interval = StockService.StockQuoteInterval.HOURLY;
+            List<StockQuote> hourlyQuoteList = stockService.getStockQuoteList(symbol, calFromDate, calUntilDate, interval);
+
+            System.out.printf("%n");
+
+            for (StockQuote temp : hourlyQuoteList) {
+                System.out.printf("The price of stock %s at time %tD %tR is %.2f %n", temp.getStockSymbol(), temp.getDateRecorded(), temp.getDateRecorded(), temp.getStockPrice());
+            }
 
-        for (StockQuote temp : hourlyQuoteList) {
-            System.out.printf("The price of stock %s at time %tD %tR is %.2f %n", temp.getStockSymbol(), temp.getDateRecorded(), temp.getDateRecorded(), temp.getStockPrice());
         }
+        catch (StockServiceException exception) {
+            System.out.printf("An exception was caught: " + exception.getMessage());
+        }
+
```

Comment:
Nice clean up here, but maybe make it even cleaner by breaking the long flow of code into discreet methods.